### ISSUE TITLE
UTF-8 characters inside xpath expression

### ISSUE THREAD

**Issue by yanshiyason** [mischov/meeseeks#95]
Hi, I was trying to use a xpath like this:

```
xpath("//span[contains(., '胡麻油最高')]"))
```

But received this error:

```
** (ArgumentError) argument error
    :erlang.list_to_binary([53, 12388, 26143, 12398, 12358, 12385])
    (meeseeks 0.15.0) src/xpath_expression_parser.yrl:433: :xpath_expression_parser.literal_expr/1
    (meeseeks 0.15.0) src/xpath_expression_parser.yrl:203: :xpath_expression_parser.yeccpars2_32/7
    (meeseeks 0.15.0) /usr/local/Cellar/erlang/22.3.2/lib/erlang/lib/parsetools-2.1.8/include/yeccpre.hrl:57: :xpath_expression_parser.yeccpars0/5
    (meeseeks 0.15.0) lib/meeseeks/selector/xpath/parser.ex:7: Meeseeks.Selector.XPath.Parser.parse_expression/1
    (meeseeks 0.15.0) lib/meeseeks/selector/xpath.ex:10: Meeseeks.Selector.XPath.compile_selectors/1
    (meeseeks 0.15.0) expanding macro: Meeseeks.XPath.xpath/1
    test/opa/goma_abura/parsers/search_parser_test.exs:2: OPA.GomaAbura.Parsers.SearchParserTest.parse_search_result/2
```

Do you have any suggestions for a quick fix? I was going to start diving into the meeseeks source code now..

**Comment by yanshiyason** [comment]
:unicode.characters_to_binary("胡麻油")

Seems to be the way to fix it.. not sure if there would be any unforseen side effects..

**Comment by mischov** [comment]
Hi @yanshiyason, thank you for your report and PR! I am reviewing now and will hopefully either get things merged or get you feedback asap.

One question that comes to my mind is if the XPath 1.0 spec prohibits unicode in any of the places you switched to `unicode:characters_to_binary`? I don't know that it'd matter even if it does, but I'd at least like to know and document if it varies from expectations.

**Comment by yanshiyason** [comment]
Hi mischov! Thank you for your quick reply!

I found 15 hits when searching for  "unicode" in this document: https://www.w3.org/TR/1999/REC-xpath-19991116/

and it seems to be supported in general:

> A single character in XPath thus corresponds to a single Unicode abstract character

**Comment by mischov** [comment]
Yeah, I just saw the same thing.

I still need to run a couple tests, but I think everything is going to be good to go.

**Comment by mischov** [comment]
Merged. I am going to leave the issue open until I cut a release including this fix which I hope will be today.

Thank you again!

**Comment by mischov** [comment]
@yanshiyason Released in v0.15.1.

**Comment by yanshiyason** [comment]
@mischov Thank you so much for the speedy response! Much appreciated!
